subworkflows/local/bam_variant_calling_sentieon_haplotyper/main.nf

```diff
@@ -0,0 +1,155 @@
+    intervals_bed_combined       // channel: [mandatory] intervals/target regions in one file unzipped, no_intervals.bed if no_intervals
+    skip_haplotyper_filter       // boolean: [mandatory] [default: false] skip haplotyper filter
+    joint_germline               // boolean: [mandatory] [default: false] joint calling of germline variants
+    sentieon_haplotyper_out_format // channel: [mandatory] value channel with string
+
+    main:
+    versions = Channel.empty()
+
+    gvcf = Channel.empty()
+    vcf = Channel.empty()
+    genotype_intervals = Channel.empty()
+
+    // Combine cram and intervals for spread and gather strategy
+    cram_intervals_for_sentieon = cram.combine(intervals)
+        // Move num_intervals to meta map
+        .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals, intervals_name:intervals.simpleName], cram, crai, intervals ] }
+
+    SENTIEON_HAPLOTYPER(
+        sentieon_haplotyper_out_format,
+        cram_intervals_for_sentieon,
+        fasta,
+        fasta_fai,
+        dbsnp,
+        dbsnp_tbi)
+
+    versions = versions.mix(SENTIEON_HAPLOTYPER.out.versions)
+
+    if (joint_germline) {
+        genotype_intervals = SENTIEON_HAPLOTYPER.out.gvcf
+            .join(SENTIEON_HAPLOTYPER.out.gvcf_tbi, failOnMismatch: true)
+            .join(cram_intervals_for_sentieon, failOnMismatch: true)
+            .map{ meta, gvcf, tbi, cram, crai, intervals -> [ meta, gvcf, tbi, intervals ] }
+    }
+
+    // Figure out if using intervals or no_intervals
+    SENTIEON_HAPLOTYPER.out.vcf.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_vcf_branch}
+
+
+    SENTIEON_HAPLOTYPER.out.vcf_tbi.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_vcf_tbi_branch}
+
+    SENTIEON_HAPLOTYPER.out.gvcf.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_gvcf_branch}
+
+    SENTIEON_HAPLOTYPER.out.gvcf_tbi.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_gvcf_tbi_branch}
+
+    // VCFs
+    // Only when using intervals
+    MERGE_SENTIEON_HAPLOTYPER_VCFS(
+        haplotyper_vcf_branch.intervals.map{ meta, vcf ->
+            new_meta = [
+                            id:             meta.sample,
+                            num_intervals:  meta.num_intervals,
+                            patient:        meta.patient,
+                            sample:         meta.sample,
+                            sex:            meta.sex,
+                            status:         meta.status,
+                            variantcaller:  "sentieon_haplotyper"
+                        ]
```

Comment:
Can we stop using new_meta, as we do in other subworkflows?

Reply:
we can just add the new id to meta

Reply:
`bam_variant_calling_sentieon_haplotyper/main.nf` is basically just an adopted copy of how `bam_variant_calling_haplotypecaller/main.nf` looked when I started out implementing this.

If you made any tricks or improvements to `bam_variant_calling_sentieon_haplotyper/main.nf` in the meantime which you would also like to see in `bam_variant_calling_haplotypecaller/main.nf`, you are very welcome to implement them.

Reply:
I was sure this was achieve before, but no worries.